src/Microsoft.Performance.SDK.Runtime/Discovery/AssemblyLoaderBase.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.Performance.SDK.Runtime.Discovery
+{
+    /// <summary>
+    ///     Base class for assembly loaders.
+    /// </summary>
+    public abstract class AssemblyLoaderBase
+        : IAssemblyLoader
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="AssemblyLoaderBase"/>
+        ///     class.
+        /// </summary>
+        protected AssemblyLoaderBase()
+        {
+        }
```

Comment:
nit: for an abstract class, the compiler will automatically generate a protected and empty .ctor